chainer/datasets/transform_dataset.py

```diff
@@ -0,0 +1,43 @@
+from chainer.dataset import dataset_mixin
+
+
+class TransformDataset(dataset_mixin.DatasetMixin):
+
+    """Dataset that indexes the base dataset and transforms the data.
+
+    This dataset wraps the base dataset by modifying the behavior of the base
+    dataset's :meth:`__getitem__`. Arrays returned by :meth:`__getitem__` of
+    the base dataset with integer as an argument are transformed by the given
+    function :obj:`transform`.
+
+    The function :obj:`transform` takes, as an argument, :obj:`in_data`, which
+    is the output of the base dataset's :meth:`__getitem__`, and returns
+    the transformed arrays as output. Please see the following example.
+
+    >>> from chainer.datasets import get_mnist
+    >>> from chainer.datasets import TransformDataset
+    >>> dataset, _ = get_mnist()
+    >>> def transform(in_data):
+    >>>     img, label = in_data
+    >>>     img -= 0.5  # scale to [-0.5, -0.5]
+    >>>     return img, label
+    >>> dataset = TransformDataset(dataset, transform)
+
+    Args:
+        dataset: The underlying dataset. The index of this dataset corresponds
```

Comment:
The dataset object given here should have `__len__` method besides `__getitem__` method because `len(self._dataset)` is called in the `__len__` method of this TransoformDataset. So I think it would be better to describe explicitly about what are required for "The underlying dataset". As for `__getitem__` method of the underlying dataset, it has been already mentioned in the beginning of this docstring, so please add an explanation about `__len__` method here or the first paragraph of the docstring.